cli/src/main/java/com/liferay/blade/server/LayeredModulePathFactory.java

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.liferay.blade.server;
+
+import com.liferay.blade.cli.util.FileUtil;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+
+import java.nio.file.Files;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * @author Brian Stansberry
+ */
+public class LayeredModulePathFactory {
+
+	/**
+	 * Load the overlays for each layer.
+	 *
+	 * @param layeringRoot the layer root
+	 * @param path the module path
+	 */
+	public static void loadOverlays(File layeringRoot, List<File> path) {
+		File overlays = new File(layeringRoot, _OVERLAYS);
+
+		if (FileUtil.exists(overlays)) {
+			File refs = new File(overlays, _OVERLAYS);
+
+			if (FileUtil.exists(refs)) {
+				try {
+					for (String overlay : readRefs(refs)) {
+						File root = new File(overlays, overlay);
+
+						path.add(root);
+					}
+				}
+				catch (IOException ioe) {
+					throw new RuntimeException(ioe);
+				}
+			}
+		}
+
+		path.add(layeringRoot);
+	}
+
+	public static boolean readLine(InputStream is, StringBuffer buffer) throws IOException {
+		buffer.setLength(0);
+		int c;
+
+		while (true) {
+			c = is.read();
+
+			switch (c) {
+				case '\t':
+				case '\r':
+
+					break;
+				case -1: return false;
+				case '\n': return true;
+				default: buffer.append((char)c);
+			}
+		}
+	}
+
+	public static List<String> readRefs(File file) throws IOException {
+		if (FileUtil.notExists(file)) {
+			return Collections.emptyList();
+		}
+
+		InputStream is = Files.newInputStream(file.toPath());
```

Comment:
Can we wrap this instead as
```
try (InputStream is = Files.newInputStream(file.toPath()) {
	return readRefs(is);
}
```
That way we don't need the finally block

Reply:
ok